src/LibraryViewExtensionMSWebBrowser/Handlers/IconResourceProvider.cs

```diff
@@ -0,0 +1,280 @@
+        /// <returns></returns>
+        public string GetResourceAsString(string url, out string extension)
+        {
+            //sometimes the urls have "about:" added to them - remove this
+            //and do it before checking cache.
+            if (url.StartsWith("about:"))
+            {
+                url = url?.Replace("about:", string.Empty);
+            }
+
+            if (!String.IsNullOrEmpty(url) && urlToBase64DataCache.ContainsKey(url))
+            {
+                var cachedData = urlToBase64DataCache[url];
+                extension = cachedData.Item2;
+                return cachedData.Item1;
+            }
+           
+            //because we modify the spec the icon urls may have been replaed by base64 enoded images
+            //if thats the case, no need to look them up again from disk, just return the string.
+            if (!String.IsNullOrEmpty(url) && url.Contains("data:image/"))
+            {
+                var match = Regex.Match(url, @"data:(?<type>.+?);base64,(?<data>.+)");
+                var base64Data = match.Groups["data"].Value;
+                var contentType = match.Groups["type"].Value;
+                //image/png -> png
+                extension = contentType.Split('/').Skip(1).FirstOrDefault();
+                urlToBase64DataCache.Add(url, (base64Data, extension));
+                return base64Data;
+            }
+          
+
+            var base64String = string.Empty;
+            extension = "png";
+            //Create IconUrl to parse the request.Url to icon name and path.
+            if (String.IsNullOrEmpty(url))
+            {
+                return string.Empty;
+            }
+
+            //before trying to create a uri we have to handle resources that might 
+            //be embedded into the resources.dlls
+            //these paths will start with ./dist
+            if (url.StartsWith(@"./dist"))
+            {
+                //make relative url a full uri
+                var urlAbs = url.Replace(@"./dist", @"http://localhost/dist");
+                var ext = string.Empty;
+                var stream = embeddedDllResourceProvider.GetResource(urlAbs, out ext);
+                if (stream != null)
+                {
+                    extension = ext;
+                    base64String = GetIconAsBase64(stream, ext);
+                }
+            }
+            else
+            {
+                //TODO check if absolute instead of using try/catch
+                try
+                {
+                    var icon = new IconUrl(new Uri(url));
+                    base64String = GetIconAsBase64(icon, out extension);
+
+                }
+                catch (Exception e)
+                {
+                    //look in resources for registered path and just use the stream directly
+                    var resourceDict = this.customization.Resources.ToDictionary(kvp => kvp.Key, kvp => kvp.Value);
```

Comment:
some hosts register streams directly so theres no need to look on disk, we can just return the stream.